packages/paste-core/components/form-pill-group/src/FormPillGroup.tsx

```diff
@@ -39,26 +45,60 @@ export interface FormPillGroupProps
    * @memberof FormPillGroupProps
    */
   display?: "flex" | "inline-flex";
+  size?: FormPillGroupSizeVariant;
+  setPillFontSize: Dispatch<SetStateAction<FontSize | undefined>>;
+}
+
+/** The response from the calculateSizeStyles function. */
+type CalculateSizeStylesResponse = Pick<BoxProps, "columnGap" | "rowGap" | "fontSize">;
+
+/**
+ * Calculates the styles for the FormPillGroup based on the size prop.
+ *
+ * Exported only to be used in tests.
+ * @private
+ */
+export function calculateSizeStyles(size: FormPillGroupSizeVariant): CalculateSizeStylesResponse {
+  if (size === "L") {
+    return {
+      columnGap: "space30",
+      rowGap: "space30",
+      fontSize: "fontSize30",
+    };
+  }
+  return {
+    columnGap: "space20",
+    rowGap: "space20",
+    fontSize: undefined,
+  };
```

Comment:
This is a great idea but not as easy to maintain. For variant mappings we normally use a record and spread the styles for the variant. 

You can find the usage to spread [here](https://github.com/twilio-labs/paste/blob/0d311e0a0b0bd57a78a7f92932bfa882a8d8594e/packages/paste-core/components/button/src/DestructiveButton.tsx#L59) and the definitions [here](https://github.com/twilio-labs/paste/blob/0d311e0a0b0bd57a78a7f92932bfa882a8d8594e/packages/paste-core/components/button/src/styles.ts#L41).

Normally they are in the same file but with so many button variants in this example it is split out.

Reply:
@krisantrobus I think this is just a plain better idea than what I've done. I'm going to adopt this pattern in my PR and make that change. I appreciate the feedback.